src/foundry/common/data/data.mjs/userData.d.ts

```diff
@@ -0,0 +1,62 @@
+import DocumentData from '../../abstract/data.mjs';
+import { _validatePermissions } from '../fields.mjs';
+import * as fields from '../fields.mjs';
+import * as documents from '../../documents.mjs';
+
+interface UserDataSchema extends DocumentSchema {
+  _id: typeof fields.DOCUMENT_ID;
+  avatar: typeof fields.IMAGE_FIELD;
+  character: fields.ForeignDocumentField<{ type: typeof documents.BaseActor; required: false }>;
+  color: DocumentField<Omit<typeof fields.COLOR_FIELD, 'required'> & { required: true }>;
+  hotbar: DocumentField<{
+    type: Object;
+    required: true;
+    default: Record<never, never>;
+    validate: typeof _validateHotbar;
+    validationError: 'Invalid User hotbar data structure';
+  }>;
+  name: typeof fields.REQUIRED_STRING;
+  password: typeof fields.BLANK_STRING;
+  permissions: DocumentField<
+    Omit<typeof fields.DOCUMENT_PERMISSIONS, 'default' | 'validate'> & {
+      default: Record<never, never>;
+      validate: typeof _validatePermissions;
+    }
+  >;
+  role: DocumentField<{
+    type: Number;
+    required: true;
+    nullable: false;
+    default: typeof CONST.USER_ROLES.PLAYER;
+  }>;
+  flags: typeof fields.OBJECT_FIELD;
+}
+
+/**
+ * Validate the structure of the User hotbar object
+ * @param bar - The attempted hotbar data
+ */
+declare function _validateHotbar(bar: unknown): asserts bar is Record<number | string, string>;
+
+interface UserDataProperties {
+  _id: string | null;
+  avatar?: string;
+  character: string | null;
+  color: string;
+  hotbar: Record<number | string, string>;
+  name: string;
+  password: '';
```

Comment:
I don't think the type should be the empty string. That would prevent any other passwords.